### PR TITLE
Fix test parameters in ChainerX math tests

### PR DESCRIPTION
**Pull request by niboshi**
Skips backward/double-backward tests for inputs that results in non-finite outputs.

Differentiable domains are different among inverse trigonometric functions.

### CODE REVIEW

**Comment by niboshi** [comment]
This PR blocks #6947.

File: tests/chainerx_tests/unit_tests/routines_tests/test_math.py [hunk 1]
```diff
@@ -21,8 +23,11 @@ def __exit__(self, *args):
 
 class UnaryMathTestBase(object):
 
+    input = None
```
Comment (asi1024) [inline tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: This fix is for the tests for array-scalar binary operation, but this class is also used for other functions which take a single array for the argument. How about defining a new 
derived class for array-scalar tests?

File: tests/chainerx_tests/unit_tests/routines_tests/test_math.py [hunk 2]
```diff
@@ -1780,7 +1810,7 @@ def func(self, xp, a):
         return xp.sqrt(a)
 
 
-_trignometric_hyperpolic_params = \
+_trigonometric_hyperpolic_params = \
```
Comment (hvy) [inline tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: A nitpick but should it be "hyperbolic"?

File: tests/chainerx_tests/unit_tests/routines_tests/test_math.py (hunk 1 above)
Comment (niboshi) [reply tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: I don't necessarily understand the point, but `self.input` is already used in thi class. Such refactoring could be done separately.

File: tests/chainerx_tests/unit_tests/routines_tests/test_math.py (hunk 2 above)
Comment (niboshi) [reply tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: Fixed.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by imos** [comment]
(PFN CI failed because master is updated just after the previous request.)
pfnCI, please test.

**Comment by imos** [comment]
pfnCI, please test
(PFN CI failed because master is updated just after the previous request.)

**Comment by asi1024** [comment]
LGTM.

**Pull request merged**